src/Components/Resource/BadgesList.tsx

```diff
@@ -1,60 +1,33 @@
-import { useState, useEffect } from "react";
-import { getAnyFacility } from "../../Redux/actions";
-import { useDispatch } from "react-redux";
+/* eslint-disable react-hooks/rules-of-hooks */
 import { SHIFTING_FILTER_ORDER } from "../../Common/constants";
+import routes from "../../Redux/api";
+import useQuery from "../../Utils/request/useQuery";
 
 export default function BadgesList(props: any) {
   const { appliedFilters, FilterBadges } = props;
-  const [orginFacilityName, setOrginFacilityName] = useState("");
-  const [approvingFacilityName, setApprovingFacilityName] = useState("");
-  const [assignedFacilityName, setAssignedFacilityName] = useState("");
-  const dispatch: any = useDispatch();
 
-  useEffect(() => {
-    async function fetchData() {
-      if (!appliedFilters.origin_facility) return setOrginFacilityName("");
-      const res = await dispatch(
-        getAnyFacility(appliedFilters.origin_facility, "origin_facility_name")
-      );
-      setOrginFacilityName(res?.data?.name);
-    }
-    fetchData();
-  }, [dispatch, appliedFilters.origin_facility]);
+  const originFacilityData =
+    appliedFilters.origin_facility &&
+    useQueryHook(appliedFilters.origin_facility);
+  const approvingFacilityData =
+    appliedFilters.approving_facility &&
+    useQueryHook(appliedFilters.approving_facility);
+  const assignedFacilityData =
+    appliedFilters.assigned_facility &&
+    useQueryHook(appliedFilters.assigned_facility);
 
-  useEffect(() => {
-    async function fetchData() {
-      if (!appliedFilters.approving_facility)
-        return setApprovingFacilityName("");
-      const res = await dispatch(
-        getAnyFacility(
-          appliedFilters.approving_facility,
-          "approving_facility_name"
-        )
-      );
-      setApprovingFacilityName(res?.data?.name);
-    }
-    fetchData();
-  }, [dispatch, appliedFilters.approving_facility]);
-
-  useEffect(() => {
-    async function fetchData() {
-      if (!appliedFilters.assigned_facility) return setAssignedFacilityName("");
-      const res = await dispatch(
-        getAnyFacility(
-          appliedFilters.assigned_facility,
-          "assigned_facility_name"
-        )
-      );
-      setAssignedFacilityName(res?.data?.name);
-    }
-    fetchData();
-  }, [dispatch, appliedFilters.assigned_facility]);
+  function useQueryHook(facilityId: any) {
+    const { data } = useQuery(routes.getAnyFacility, {
+      pathParams: { id: facilityId },
+    });
+    return { data };
+  }
 
   const getDescShiftingFilterOrder = (ordering: any) => {
     let desc = "";
-    SHIFTING_FILTER_ORDER.map((item: any) => {
+    SHIFTING_FILTER_ORDER.forEach((item) => {
       if (item.text === ordering) {
-        desc = item.desc;
+        desc = item.desc!;
       }
     });
     return desc;
```

Comment:
You can do a `.find` instead of declaring a empty variable, iterating and assigning, and then returning right?